src/simulator/std.rs

```diff
@@ -29,3 +29,62 @@ pub fn sleep<T: crate::traits::Bag>(
         t_next
     }
 }
+
+/// Closure for waiting for an event in real-time simulation.
+/// It calculates the next real-time and virtual-time based on the given time scale and sleeps until the next state transition.
+/// If the maximum jitter is provided, it checks if the actual time exceeds the future time and panics if it does.
+/// It also calls the input handler function with the duration between the current and next real-time.
+///
+/// # Arguments
+///
+/// * `t_start` - The starting virtual time.
+/// * `time_scale` - The time scale factor.
+/// * `max_jitter` - The maximum allowed jitter duration.
+/// * `input_handler` - The function to handle the input event.
```

Comment:
```suggestion
/// - * `t_start` - The virtual time at the beginning of the simulation.
/// - * `time_scale` - The time scale factor between virtual and wall-clock time.
/// - * `max_jitter` - The maximum allowed jitter duration. If `None`, no jitter check is performed.
/// - * `input_handler` - The function to handle incoming external events. This function expects two arguments:
///    - `duration: [`Duration`]` - Maximum duration of the time interval to wait for external events.
The input handler function may return earlier if an input event is received.
Note, however, that it must **NOT** return after, as it would result in an incorrect real-time implementation.
///    - `input_ports: &mut T` - Mutable references to the input ports of the top-most model under simulation.
```